python/ray/ml/predictors/integrations/lightgbm/lightgbm_predictor.py

```diff
@@ -31,27 +36,29 @@ def from_checkpoint(cls, checkpoint: Checkpoint) -> "LightGBMPredictor":
                 ``LightGBMTrainer`` run.
 
         """
-        raise NotImplementedError
+        path = checkpoint.to_directory()
```

Comment:
Does this guarantee files are written to a new directory? If it re-uses the existing checkpoint dir the rmtree below could delete the checkpoint unexpectedly.

Reply:
Yes: `path = path if path is not None else _temporary_checkpoint_dir()`

so it is a temporary directory

Reply:
It should always make a temporary directory - at least that's how I understand the code. @krfricke can you confirm?